lib/libc/aarch64/gen/setjmp.S

```diff
@@ -121,30 +117,27 @@ ENTRY(longjmp)
 	ldr	REG(8), [REG(0)], #(REG_WIDTH)
 	mov	REGN(sp), REG(8)
 
+#if defined(__CHERI_PURE_CAPABILITY__) && defined(RTLD_SANDBOX)
+	/* Restore Executive mode state and general purpose registers */
+	bl	_rtld_longjmp
+	/* Restore lr */
+	ldr	clr, [c0], #CAP_WIDTH
+#else
```

Comment:
Why does setjmp do the save but longjmp not do the restore?

Reply:
Callee-saved registers are saved and restored by the trampoline. So even if longjmp does the restore, the registers would be overwritten when longjmp returns. The restore has to be done by _rtld_longjmp, which overwrites the trusted stack frame with the correct register values to be restored.